neuraxle/base.py

```diff
@@ -752,6 +753,30 @@ def get_hyperparams_space(self) -> HyperparameterSpace:
         """
         return self.hyperparams_space
 
+    def apply_method(self, method: Callable, *kargs) -> 'BaseStep':
+        """
+        Apply a method to a step and its children.
+
+        :param method: method to call with self
+        :param kargs: any additional arguments to be passed to the method
+        :return:
+        """
+        method(self, *kargs)
+        return self
+
+    def apply(self, method_name: str, *kargs) -> 'BaseStep':
+        """
+        Apply a method to a step and its children.
+
+        :param method_name: method name that need to be called on all steps
+        :param kargs: any additional arguments to be passed to the method
+        :return:
+        """
+        if hasattr(self.__class__, method_name) and callable(getattr(self.__class__, method_name)):
```

Comment:
I believe you should do `f hasattr(self, method_name)`, not `f hasattr(self.__class__, method_name)`. Im not sure. Same for `callable` after in the if.

Reply:
right someone could have deleted the attr. it works with self. thanks !